emu/src/arm7tdmi.rs

```diff
@@ -13,7 +13,7 @@ use crate::{cpsr::Cpsr, cpu::Cpu};
 /// Contains the 16 registers for the CPU, latest (R15) is special because
 /// is the program counter.
 #[derive(Default)]
-struct Registers([u32; 16]);
+pub struct Registers([u32; 16]);
```

Comment:
Why do we sometimes use `pub` and sometimes `pub(crate)`?

I ask because I'm a rust noob :)

Reply:
https://doc.rust-lang.org/reference/visibility-and-privacy.html
Basically `pub` is public also for other crates (if you expose your crate)
`pub(crate)` makes public only within the same crate but for other modules